opal/mca/btl/ofi/btl_ofi_module.c

```diff
@@ -277,20 +277,32 @@ int mca_btl_ofi_finalize (mca_btl_base_module_t* btl)
 
     assert(btl);
 
+    /* clear the rcache */
+    if (ofi_btl->rcache) {
+        mca_rcache_base_module_destroy (ofi_btl->rcache);
+        ofi_btl->rcache = NULL;
+    }
+
+    /* For a standard ep, we need to close the ep first. */
+    if (NULL != ofi_btl->ofi_endpoint && !ofi_btl->is_scalable_ep) {
+        fi_close(&ofi_btl->ofi_endpoint->fid);
+        ofi_btl->ofi_endpoint = NULL;
+    }
+
     /* loop over all the contexts */
     for (i=0; i < ofi_btl->num_contexts; i++) {
         mca_btl_ofi_context_finalize(&ofi_btl->contexts[i], ofi_btl->is_scalable_ep);
     }
     free(ofi_btl->contexts);
 
-    if (NULL != ofi_btl->av) {
-        fi_close(&ofi_btl->av->fid);
-    }
-
     if (NULL != ofi_btl->ofi_endpoint) {
         fi_close(&ofi_btl->ofi_endpoint->fid);
     }
 
+    if (NULL != ofi_btl->av) {
+        fi_close(&ofi_btl->av->fid);
```

Comment:
Ordering problem again it seems. Good to have a comment here for that.

Reply:
I added a brief comment. Should I instead have added a block comment ?